tools/automation.py

```diff
@@ -0,0 +1,316 @@
+"""
+    Create a script that will convert mentor's data from mentors.xlsx to mentors.yml file
+
+    Run script:
+        \tools>run_mentors_automation.bat
+        Enter arguments for Python script:mentors.xlsx mentors.yml a
+"""
+#!/usr/bin/env python
+
+import sys
+import logging
+import textwrap
+import re
+from enum import Enum
+import pandas as pd
+from ruamel.yaml import YAML
+from ruamel.yaml.scalarstring import LiteralScalarString
+
+SHEET_NAME = "Form Responses 1"
+TELEGRAM_WEB_SITE = '//t.'
+SOCIAL_MEDIA = ['linkedin', 'twitter', 'github', 'medium', 'youtube',\
+                'instagram', TELEGRAM_WEB_SITE, 'meetup', 'slack', 'facebook']
+WEBSITE = 'website'
+TELEGRAM = 'telegram'
+
+# Indexes for creating yaml sequences of data
+AREAS_START_INDEX = 13
+AREAS_END_INDEX = 17
+FOCUS_START_INDEX = 18
+FOCUS_END_INDEX = 22
+PROG_LANG_START_INDEX = 23
+PROG_LANG_END_INDEX = 27
+
+TYPE_AD_HOC = "ad hoc"
+TYPE_LONG_TERM = "long-term"
+TYPE_BOTH = "both"
+IMAGE_FILE_PATH = "assets/images/mentors/"
+IMAGE_SUFFIX = ".jpeg"
+
+class WriteMode(Enum):
+    WRITE = "w"
+    APPEND = "a"
+
+
+def get_social_media_links(*links_args):
+    """
+    Prepare mentor's social media links for yaml network sequence.
+    """
+    network_list = []
+    links_str = ""
+
+    for item in links_args:
+        if not pd.isna(item):
+            links_str += item + " "
+
+    social_media_links_list = links_str.split()
+
+    for link in social_media_links_list:
+        found = 0
+        for name in SOCIAL_MEDIA:
+            if link.find(name) != -1:
+                if name == TELEGRAM_WEB_SITE:
+                    network_list.append({TELEGRAM: link})
+                else:
+                    network_list.append({name: link})
+                found = 1
+                break
+        if found == 0 and len(link):
```

Comment:
Done

Reply:
```
def fallback_link(link):
    """
        Use "webpage" keyword as fallback in case of unknown link.
     """
    return {WEBSITE: link}
```

```
def get_social_media_links(*links_args):
    ...
       if found == 0:
            network_list.append(fallback_link(link))
```

Reply:
Also extrcted the part of code where the *args are checked no to be nan or empty str to a new function:
```
def strings_to_list(*string_args):
    """
    Clean strings and create to list
    """
    data_str = ""

    for item in string_args:
        if not pd.isna(item) and len(item):
            data_str += item + " "

    return data_str.split()
```

```
def get_social_media_links(*links_args):
    """
    Prepare mentor's social media links for yaml network sequence.
    """
    network_list = []
    social_media_links_list = strings_to_list(*links_args)

    for link in social_media_links_list:

...
```